src/libs/nav.ts

```diff
@@ -7,6 +8,38 @@ import { useOnMount, useStore } from 'src/libs/react-utils';
 import { routeHandlersStore } from 'src/libs/state';
 import * as Utils from 'src/libs/utils';
 
+// TODO: add all used nav key names here and switch consumers to use the safe const values here
+/**
+ * union type (enum-like) of safe nav key names
+ */
+export type TerraNavKey = 'billing' | 'workspace-dashboard' | 'workspaces';
+
+export type TerraNavKeyLookup = { [key in TerraNavKey]: key };
+
+// TODO: fill out this value lookup to reflect full list of key names once available above
+/**
+ * allowed values for nav keys in a lookup table
+ * Note: use terraNavKey type helper method below instead of this const.
+ *   Typescript currently allows unsafe object['not-an-enum-key-name'] indexing
+ *   lookups into this otherwise type-safe construct.
+ */
+export const terraNavKeyValue: TerraNavKeyLookup = {
+  billing: 'billing',
+  'workspace-dashboard': 'workspace-dashboard',
+  workspaces: 'workspaces',
+};
+
+/**
+ * type helper for nav keys to better enforce key names with snake-case
+ * @param key
+ */
+export const terraNavKey = (key: TerraNavKey) => enforceType<TerraNavKey>(key);
```

Comment:
```suggestion
export const terraNavKey = enforceType<TerraNavKey>;
```

This could be simplified. I find the simpler version clearer. As is, there's no reason to use `enforceType` because the type is already enforced by the type declaration on the `key` parameter.

Reply:
I hit an ergonomic barrier where just using enforceType didn't give the IDE enough direct TS goodness to fully restrict the auto-suggest values of key arg to the valid key values... so I had to re-introduce the arg typing here... I kinda like that enforceType is self-documenting even if the end result here is not as crisp/concise as we hoped.

Reply:
In that case, I still think the `enforceType` is useless here: it's an extra function call for no gain. I'd vote for something like:
```suggestion
export const enforceTerraNavKeyType = (key: TerraNavKey) => key;
```

Reply:
Looks like it depends on the editor too... VS Code autocompletes arguments for something like `const enforceMyType = enforceType<MyType>;`.